projects/kit/components/slider/test/slider-key-steps.spec.ts

```diff
@@ -0,0 +1,134 @@
+import {Component, ElementRef, ViewChild} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {FormControl, ReactiveFormsModule} from '@angular/forms';
+import {TuiKeySteps} from '@taiga-ui/kit';
+import {configureTestSuite} from '@taiga-ui/testing';
+
+import {TuiSliderModule} from '../slider.module';
+
+describe('TuiSliderKeyStepsDirective', () => {
+    @Component({
+        template: `
+            <input
+                #slider
+                tuiSlider
+                type="range"
+                [keySteps]="keySteps"
+                [max]="max"
+                [min]="min"
+                [formControl]="control"
+            />
+        `,
+    })
+    class TestComponent {
+        @ViewChild('slider', {static: true, read: ElementRef})
+        inputElRef!: ElementRef<HTMLInputElement>;
+
+        control = new FormControl(720_000);
+
+        keySteps: TuiKeySteps = [
+            [0, 5_000],
+            [100 / 3, 100_000],
+            [(100 / 3) * 2, 300_000],
+            [100, 1_000_000],
+        ];
+
+        max = 30;
+        min = 0;
+    }
+
+    let fixture: ComponentFixture<TestComponent>;
+    let testComponent: TestComponent;
+
+    configureTestSuite(() => {
+        TestBed.configureTestingModule({
+            imports: [ReactiveFormsModule, TuiSliderModule],
+            declarations: [TestComponent],
+        });
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(TestComponent);
+        testComponent = fixture.componentInstance;
+    });
+
+    describe("correctly sets initial value on native input (values satisfy input's steps)", () => {
+        it('720_000 => 26/30', () => {
+            fixture.detectChanges();
+            expect(testComponent.inputElRef.nativeElement.value).toBe('26');
+        });
+
+        const controlNativeValuesMap = [
```

Comment:
what do you think about add unit test with float values?

Reply:
sounds reasonable, I'll add it 👌

Reply:
@splincode I added unit tests with float numbers

(it has taken me a huge amount of time to calculate values by hand (to be sure that tests check right values) 🤣)